tests/back/tasks/test_tasks.py

```diff
@@ -26,36 +26,69 @@ def test_create_tabular_task():
 
 
 def test_validate_task():
-    datasetdashai_csv_created = datasetdashai_from_csv("iris.csv")
+    dashaidataset = dashaidataset_from_csv("iris.csv")
     inputs_columns = ["SepalLengthCm", "SepalWidthCm", "PetalLengthCm", "PetalWidthCm"]
     outputs_columns = ["Species"]
-    datasetdict = to_dashai_dataset(
-        datasetdashai_csv_created, inputs_columns, outputs_columns
-    )
+    name_datasetdict = "Iris"
+    datasetdict = to_dashai_dataset(dashaidataset, inputs_columns, outputs_columns)
     tipos = {"Species": "Categorico"}
     for split in datasetdict:
         datasetdict[split] = datasetdict[split].change_columns_type(tipos)
     tabular_task = TabularClassificationTask.create()
-    tabular_task.validate_dataset_for_task(datasetdict)
+    tabular_task.validate_dataset_for_task(datasetdict, name_datasetdict)
     assert True
 
 
-def test_wrong_cardinality_task():
-    with pytest.raises(ValueError):
-        datasetdashai_csv_created = datasetdashai_from_csv("iris_extra_feature.csv")
+def test_wrong_type_task():
+    with pytest.raises(TypeError):
+        datasetdashai_csv_created = dashaidataset_from_csv("iris_extra_feature.csv")
         inputs_columns = [
             "SepalLengthCm",
             "SepalWidthCm",
             "PetalLengthCm",
             "PetalWidthCm",
         ]
         outputs_columns = ["Species", "StemCm"]
+        name_datasetdict = "Iris"
         datasetdict = to_dashai_dataset(
             datasetdashai_csv_created, inputs_columns, outputs_columns
         )
         tipos = {"Species": "Categorico"}
         for split in datasetdict:
             datasetdict[split] = datasetdict[split].change_columns_type(tipos)
         tabular_task = TabularClassificationTask.create()
-        tabular_task.validate_dataset_for_task(datasetdict)
+        tabular_task.validate_dataset_for_task(datasetdict, name_datasetdict)
+    assert True
+
+
+def test_prepare_task():
+    datasetdashai_csv_created = dashaidataset_from_csv("iris.csv")
+    inputs_columns = ["SepalLengthCm", "SepalWidthCm", "PetalLengthCm", "PetalWidthCm"]
+    outputs_columns = ["Species"]
+    name_datasetdict = "Iris"
+    datasetdict = to_dashai_dataset(
+        datasetdashai_csv_created, inputs_columns, outputs_columns
+    )
+    tabular_task = TabularClassificationTask.create()
+    datasetdict = tabular_task.prepare_for_task(datasetdict)
+    tabular_task.validate_dataset_for_task(datasetdict, name_datasetdict)
+    assert True
```

Comment:
To test a function that does not raises anything, you could use a try/catch block. In try, you execute the code. If the code fails, pytest.fail is executed, which reports the error:

```suggestion
    try:
        tabular_task.validate_dataset_for_task(datasetdict, name_datasetdict)
    except Exception as e:
        pytest.fail(f"Unexpected error in test_prepare_task: {repr(e)}")
```